python_modules/libraries/dagster-dg-cli/dagster_dg_cli/cli/api/API_IMPLEMENTATION_PLAN.md

```diff
@@ -0,0 +1,413 @@
+
+# Future verbs
+dg api deployment view <name> [--json]
+dg api deployment create --name <name> --type <serverless|hybrid>
+dg api deployment update <name> [--settings <file>]
+dg api deployment delete <name>
+```
+
+**GraphQL**: `fullDeployments`, `currentDeployment`, `deploymentByName`
+
+### 2. **secret**
+
+```bash
+dg api secret list [--location <name>] [--scope <deployment|organization>] [--json]
+dg api secret view <name> [--location <name>] [--json]
+dg api secret create --name <name> --value <value> [--location <name>] [--scope <scope>]
+dg api secret update <name> --value <value> [--location <name>]
+dg api secret delete <name> [--location <name>]
+```
+
+**GraphQL**: `secretsOrError`
+
+### 3. **agent**
+
+```bash
+dg api agent list [--json]
+dg api agent view <id> [--json]
+# Agents are typically managed via configuration, not CRUD operations
+```
+
+**GraphQL**: `agents`
+
+### 4. **run**
+
+```bash
+dg api run list [--status <status>] [--pipeline <name>] [--limit <n>] [--json]
+dg api run view <run-id> [--json]
+dg api run create --pipeline <name> [--config <file>] [--tags <key=value>]
+dg api run terminate <run-id>
+dg api run delete <run-id>
+```
+
+**GraphQL**: `runsOrError`, `pipelineRunsOrError`, `runOrError`
+
+### 5. **asset**
+
+```bash
+dg api asset list [--prefix <path>] [--limit <n>] [--json]
+dg api asset view <asset-key> [--json]
+# Assets are typically managed through code, not direct API
+```
+
+**GraphQL**: `assetsOrError`, `assetNodes`, `assetOrError`
+
+---
+
+## **Tier 2: Cloud Management**
+
+### 6. **user**
+
+```bash
+dg api user list [--json]
+dg api user view <user-id> [--json]
+dg api user create --email <email> --role <role>
+dg api user update <user-id> --role <role>
+dg api user delete <user-id>
+```
+
+**GraphQL**: `usersOrError`
+
+### 7. **team**
+
+```bash
+dg api team list [--json]
+dg api team view <team-id> [--json]
+dg api team create --name <name> [--members <user-id1,user-id2>]
+dg api team update <team-id> [--add-member <user-id>] [--remove-member <user-id>]
+dg api team delete <team-id>
+```
+
+**GraphQL**: `teamPermissions`
+
+### 8. **alert-policy**
+
+```bash
+dg api alert-policy list [--json]
+dg api alert-policy view <policy-id> [--json]
+dg api alert-policy create --name <name> --config <file>
+dg api alert-policy update <policy-id> --config <file>
+dg api alert-policy delete <policy-id>
+```
+
+**GraphQL**: `alertPolicies`, `alertPolicyById`
+
+### 9. **code-location**
+
+```bash
+dg api code-location list [--json]
+dg api code-location view <location-name> [--json]
+dg api code-location create --name <name> --image <image> --config <file>
+dg api code-location update <location-name> --image <image> [--config <file>]
+dg api code-location delete <location-name>
+```
+
+**GraphQL**: `repositoriesOrError`, `repositoryOrError`
+
+### 10. **workspace**
+
+```bash
+dg api workspace list [--json]
+dg api workspace view [--json]
+dg api workspace update --config <file>
```

Comment:
do you want to keep this 'workspace' terminology? i thought we were de-emphasizing it. (maybe just code-locations?)